File: app/src/main/java/com/gomart/guildbuddy/network/interfaces/GuildService.java
package com.gomart.guildbuddy.network.interfaces;

import com.gomart.guildbuddy.network.GetGuildMembersResponse;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by glaubermartins on 2016-11-24.
 */

public interface GuildService {

    @GET("wow/guild/{realm}/{guildName}")
    Call<GetGuildMembersResponse> getGuildMembers(@Path("realm") String realm, @Path("guildName") String guildName, @Query("fields") ArrayList<String> fields, @Query("locale") String locale, @Query("apikey") String apiKey);

}

File: app/src/main/java/com/gomart/guildbuddy/network/services/GuildService.kt
package com.gomart.guildbuddy.network.services

import com.gomart.guildbuddy.network.GuildMembers
import retrofit2.Response
import retrofit2.http.*

/**
 * Created by glaubermartins on 2016-11-24.
 */
interface GuildService {
    @GET("data/wow/guild/{realm}/{guildName}/roster")
    suspend fun fetchGuildMembers(
            @Path("realm") realm: String,
            @Path("guildName") guildName: String,
            @Query("namespace") namespace: String,
            @Query("locale") locale: String,
            @Query("access_token") token: String): Response<GuildMembers>
}